custom_components/waste_collection_schedule/waste_collection_schedule/source/redland_qld_gov_au.py

```diff
@@ -0,0 +1,130 @@
+import json
+from datetime import date, timedelta
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Redland City Council (QLD)"
+DESCRIPTION = "Source for Redland City Council (QLD) rubbish collection."
+URL = "https://www.redland.qld.gov.au"
+TEST_CASES = {
+    "Mount Cotton": {
+        "suburb": "Mount Cotton",
+        "street_name": "Mount Cotton Road",
+        "street_number": "1,261",
+    },
+    "Random Redland Bay": {
+        "suburb": "Redland Bay",
+        "street_name": "Boundary Street",
+        "street_number": "1",
+    },
+    "Random Victoria Point": {
+        "suburb": "Victoria Point",
+        "street_name": "Colburn Avenue",
+        "street_number": "25",
+    },
+}
+
+HEADERS = {"user-agent": "Mozilla/5.0"}
+
+API = "https://redland.waste-info.com.au/api/v1"
+
+ICON_MAP = {
+    "waste": "mdi:trash-can",
+    "recycle": "mdi:recycle",
+    "organic": "mdi:leaf",
+}
+
+
+class Source:
+    def __init__(self, suburb, street_name, street_number):
+        self.suburb = suburb
+        self.street_name = street_name
+        self.street_number = street_number
+
+    def fetch(self):
+        suburb_id = 0
+        street_id = 0
+        property_id = 0
+        today = date.today()
+        nextyear = today + timedelta(365)
+
+        # Retrieve suburbs
+        r = requests.get(f"{API}/localities.json", headers=HEADERS)
+        data = json.loads(r.text)
+
+        # Find the ID for our suburb
+        for item in data["localities"]:
+            if item["name"] == self.suburb:
+                suburb_id = item["id"]
+                break
+
+        if suburb_id == 0:
+            return []
```

Comment:
Sorry missed some thing:
could you please raise an Exception instead of returning [] as this will be shown in the logs while just returning an empty list is not transparent why you do not get any events